DigitalOcean.API/Helpers/JsonDeserializationHelper.cs

```diff
@@ -1,15 +1,29 @@
+using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
-using RestSharp;
+using Newtonsoft.Json.Serialization;
+using System;
 
 namespace DigitalOcean.API.Helpers {
     internal class JsonDeserializationHelper {
 
-        public static T DeserializeWithRootElementName<T>(string json, string rootElement) {
-            var parsedJson = JObject.Parse(json);
-            if (rootElement == null || !parsedJson.ContainsKey(rootElement))
-                return (T)parsedJson.ToObject(typeof(T));
+        public static JsonSerializerSettings DeserializerSettings { get; private set; } = new JsonSerializerSettings() {
+            DateFormatString = "yyyy-MM-ddTHH:mm:ssZ",
+            NullValueHandling = NullValueHandling.Ignore,
+            ContractResolver = new DefaultContractResolver() { NamingStrategy = new SnakeCaseNamingStrategy() }
+        };
 
-            return (T)parsedJson[rootElement].ToObject(typeof(T));
+        public static JsonSerializer Deserializer { get; private set; } = JsonSerializer.Create(DeserializerSettings);
+
+        public static T DeserializeWithRootElementName<T>(JObject parsedJson, string rootElement, JsonSerializer deserializer = null) {
+            if (parsedJson == null) throw new ArgumentNullException(nameof(parsedJson));
```

Comment:
This should not happen for the most part. Just added for safety.
Not doing null checking at the calling side.